code/japanese/reading.py

```diff
@@ -152,29 +152,30 @@ def reading(self, expr: str) -> str:
                 if kanji[i] != reading[i]:
                     break
                 placeL = i + 1
-            if placeL == 0:
-                if placeR == 0:
+            if not placeL:
+                if not placeR:
                     out.append(f" {kanji}[{reading}]")
                 else:
                     out.append(
-                        " %s[%s]%s"
-                        % (kanji[:-placeR], reading[:-placeR], reading[-placeR:])
+                        f" {kanji[:-placeR]}"
+                        f"[{reading[:-placeR]}]"
+                        f"{reading[-placeR:]}"
                     )
             else:
-                if placeR == 0:
+                if not placeR:
                     out.append(
-                        "%s %s[%s]"
-                        % (reading[:placeL], kanji[placeL:], reading[placeL:])
+                        f"{reading[:placeL]}"
+                        f" "
```

Comment:
I think some linters will complain if a format string contains no replacements.

Reply:
Right, this string doesn't need the `f`. Removed it.